include/fbgemm/QuantUtils.h

```diff
@@ -60,7 +60,12 @@ FBGEMM_API T Quantize(
     float scale,
     int result_precision,
     bool result_is_signed = std::is_signed<T>::value) {
-  const float transformed_val = zero_point + src / scale;
+  float transformed_val = std::nearbyintf(zero_point + src / scale);
```

Comment:
@zafartahirov and I addressed this problem in: https://github.com/pytorch/pytorch/pull/22830

Reply:
we are making sure no overflow in caller site